infra/scripts/validate-version-consistency.sh

```diff
@@ -1,80 +1,131 @@
 #!/usr/bin/env bash
 
 # This script will scan through a list of files to validate that all versions are consistent with
-# - Master version (could be snapshot)
-# - Highest stable commit (latest tag)
+# - master version:  (could be snapshot)
+# - Docker images version: 'dev' on master, Highest tag on release branches 
+# - Release version: Highest stable commit. Latest tag repo wide, release candidates not included
+# Usage: ./validate-version-consistency.sh 
+# Optionaly set TARGET_MERGE_BRANCH var to the target merge branch to lint 
+#   versions against the given merge branch.
 set -e
 
+BRANCH_NAME=${TARGET_MERGE_BRANCH-$(git rev-parse --abbrev-ref HEAD)}
+# Matches (ie vMAJOR.MINOR-branch) release branch names
+RELEASE_BRANCH_REGEX="^v[0-9]+\.[0-9]+-branch$"
+
 # Determine the current Feast version from Maven (pom.xml)
-export FEAST_MASTER_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
-[[ -z "$FEAST_MASTER_VERSION" ]] && {
-  echo "$FEAST_MASTER_VERSION is missing, please check pom.xml and maven"
+export FEAST_MAVEN_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+[[ -z "$FEAST_MAVEN_VERSION" ]] && {
+  echo "$FEAST_MAVEN_VERSION is missing, please check pom.xml and maven"
   exit 1
 }
+echo "Linting Maven version: $FEAST_MAVEN_VERSION"
 
-# Determine the highest released version from Git history
-git fetch --prune --unshallow --tags || true
-FEAST_RELEASE_VERSION_WITH_V=$(git tag -l --sort -version:refname | head -n 1)
-echo $FEAST_RELEASE_VERSION_WITH_V
-
-export FEAST_RELEASE_VERSION=${FEAST_RELEASE_VERSION_WITH_V#"v"}
-echo $FEAST_RELEASE_VERSION
+# Determine Docker image version tag relative to current branch
+if [ $BRANCH_NAME = "master" ]
+then
+    # Use development version
+    FEAST_DOCKER_VERSION="develop"
+elif echo "$BRANCH_NAME" | grep -P $RELEASE_BRANCH_REGEX &>/dev/null
+then
+    # Use last release tag tagged on the release branch
+    LAST_MERGED_TAG=$(git tag -l --sort -version:refname --merged | head -n 1)
```

Comment:
`--merged` would ensure lint-version will to work if user checks out a specific commit in a vX.Y-branch release branch as it only lists tags reachable from HEAD.